app/models/task.py

```diff
@@ -1,5 +1,33 @@
 from app import db
 
-
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    task_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title=db.Column(db.String(80))
+    description=db.Column(db.String(100))
+    completed_at=db.Column(db.DateTime,nullable=True)
+    
+    goal_id = db.Column(db.Integer, db.ForeignKey('goal.goal_id'))
+    goal = db.relationship("Goal", back_populates="tasks")
+
+    
+    def to_dict(self):
+        if self.goal_id:
+            return{
+                "id": self.task_id,
+                "title": self.title,
+                "description": self.description,
+                "is_complete": self.completed_at is not None,
+                "goal_id": self.goal_id
+            }
+        return {
+                "id": self.task_id,
+                "title": self.title,
+                "description": self.description,
+                "is_complete": self.completed_at is not None,
+            }
```

Comment:
Ooo, this is a good place to DRY! Remember that the pipe operator `|` can merge two dictionaries together: 

```
goal_dict = {
    "id": self.task_id,
    "title": self.title,
    "description": self.description,
    "is_complete": self.completed_at is not None,
}

if self.goal_id:
    return goal_dict | {"goal_id": self.goal_id} 
else:
    return goal_dict
```